Userland/Libraries/LibJS/Runtime/RegExpConstructor.cpp

```diff
@@ -91,4 +179,151 @@ JS_DEFINE_NATIVE_FUNCTION(RegExpConstructor::symbol_species_getter)
     return vm.this_value();
 }
 
+// get RegExp.input, https://github.com/tc39/proposal-regexp-legacy-features#get-regexpinput
+JS_DEFINE_NATIVE_FUNCTION(RegExpConstructor::input_getter)
+{
+    auto* regexp_constructor = vm.current_realm()->intrinsics().regexp_constructor();
+
+    // 1. Return ? GetLegacyRegExpStaticProperty(%RegExp%, this value, [[RegExpInput]]).
+    return js_string(vm, regexp_constructor->legacy_static_properties().input());
+}
+
+// get RegExp.$_, https://github.com/tc39/proposal-regexp-legacy-features#get-regexp_
+JS_DEFINE_NATIVE_FUNCTION(RegExpConstructor::input_alias_getter)
+{
+    // Keep the same implementation with `get RegExp.input`
+    return input_getter(vm);
+}
+
+// SetLegacyRegExpStaticProperty( C, thisValue, internalSlotName, val ) https://github.com/tc39/proposal-regexp-legacy-features#setlegacyregexpstaticproperty-c-thisvalue-internalslotname-val-
+static void set_legacy_regexp_static_property(VM& vm, Function<void(Utf16View const&)>& property_setter, String const& new_value)
+{
+    (void)vm;
+
+    // FIXME: 1. Assert C is an object that has an internal slot named internalSlotName.
```

Comment:
This is then implied through the setter function's C++ type checking

```suggestion
    // 1. Assert C is an object that has an internal slot named internalSlotName.
```